src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -7,8 +7,32 @@
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
-    @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException(
+                    "Login cannot be null");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
```

Comment:
This check for existing user is good, but it would be better to check not only the login, but also the password. If the password is different, it could be a different user with the same login.